### DISCUSSION TITLE
Debugging Conditions

### DISCUSSION THREAD

**Discussion by WFDexter** [wixtoolset/issues#8710] [Questions] [answered]
I'm going crazy trying to get conditional installation and execution of a C# CA to work. I just keep messing around with the WXS file and different things happen but I'm not sure why. The CA was running for a while but I seem to have broken that and have no idea what I changed that impacted it.

Does anyone have suggestions for how to see what's going on with these conditions? "msiexec ... /l*xv log.txt" doesn't seem to even mention the conditions. I remember having similar problems the last time I did conditional install but it was literally years ago.

If somebody wants to actually look at this and tell me what I'm missing that would be great, but I'd love to even have some tips about how to debug this stuff.

Note that I'm still using Wix 3.11. I didn't honestly even realize it was up to version 5 and this isn't the time to try to switch even to 4. 

I have three requirements I'm trying to satisfy:

**1. If directory exists, install file to it. This is working so I think this bit is right.**

  ```
<Property Id="STC_PATH_PRESENT">
              <DirectorySearch Path="c:\mcdusa\installers" Depth="0" AssignToProperty="yes" Id="DS_STC_PATH" />
</Property>

<Component Id="CMP_STC_UPG" Guid="{f14f1b15-7a58-4b28-98f6-0fcf897ca831}">
    <Condition>
        <![CDATA[STC_PATH_PRESENT]]> 
    </Condition>
    <File Id="FIL_STC_UPG" Source="..\..\Output\vendor.exe" KeyPath="yes" />
</Component>
```
**2. If #1 directory exists, and property "CONVERT" is "YES", set on the command line, install one or the other version of a file to a different directory. Notice that one has condition "CONVERT=YES" and the other "CONVERT=NO". Each of these components are in separate Component Groups with the same Directory value.**

```
<Property Id="CONVERT" Secure="yes" >NO</Property>

<Component Id="CMP_PRE400_CMD_UPG" Guid="{1fe32293-9b5d-401d-abf6-2290710a4d30}">
    <File Id="FILE_PRE400_CMD_UPG" Source="pre400.cmd" KeyPath="yes" />
    <Condition>
        <![CDATA[STC_PATH_PRESENT And (CONVERT=NO)]]> 
        </Condition>
</Component>

<Component Id="CMP_PRE400_CMD_CON" Guid="{6773b78b-2f19-481f-b501-1a8ce046d8b1}">
    <File Id="FILE_PRE400_CMD_CON" Source="pre400.llz" Name="pre400.cmd" KeyPath="yes" />
    <Condition>
        <![CDATA[STC_PATH_PRESENT And (CONVERT=YES)]]> 
    </Condition>
</Component>
```

**3. If #1 directory exists and property CONVERT is YES, run a custom action.**

```
<InstallExecuteSequence>
    <Custom Action="CA_TodayRestartTime" Before="InstallFinalize" > STC_PATH_PRESENT And (CONVERT=YES) And (NOT Installed) </Custom>
</InstallExecuteSequence>

```

**Comment by robmen** [comment, marked as answer]
Have you tried quoting your string constants? `YES` is a Property or Directory reference. `"YES"` is a string with the characters `Y` `E` and `S` in that order.

See [MSI Conditional Statement Syntax](https://learn.microsoft.com/en-us/windows/win32/msi/conditional-statement-syntax)

**Comment by WFDexter** [comment]
That was it! Thank you so much!

I'm pretty sure I gave up on this the last time I tried it and figured out how to get the end result some other way. I did read that page you linked repeatedly but I kept not seeing the bullet point about quotes for strings.
